hazelcast/src/hazelcast/client/config.cpp

```diff
@@ -160,16 +160,75 @@ namespace hazelcast {
             client_flake_id_generator_config &
             client_flake_id_generator_config::set_prefetch_validity_duration(std::chrono::milliseconds duration) {
                 util::Preconditions::check_not_negative(duration.count(),
-                                                      "prefetchValidityMs must be non negative");
+                                                        "prefetchValidityMs must be non negative");
                 prefetch_validity_duration_ = duration;
                 return *this;
             }
 
-            int32_t client_network_config::CONNECTION_ATTEMPT_PERIOD = 3000;
+            constexpr std::chrono::milliseconds connection_retry_config::INITIAL_BACKOFF;
+            constexpr std::chrono::milliseconds connection_retry_config::MAX_BACKOFF;
+            constexpr std::chrono::milliseconds connection_retry_config::CLUSTER_CONNECT_TIMEOUT;
+            constexpr double connection_retry_config::JITTER;
```

Comment:
These aren't necessary for static constexpr variables.
```suggestion
```

Reply:
Unfortunately it does not compile that way, the errors are like:
```
Undefined symbols for architecture x86_64:
  "hazelcast::client::config::connection_retry_config::MAX_BACKOFF", referenced from:
      hazelcast::client::config::connection_retry_config::connection_retry_config() in config.cpp.o
```

Reply:
Could it be because the header `connection_retry_config.h` was not included in `config.cpp`?

Reply:
OK, they are necessary. They are not for C++17 though, that is probably the reason I thought so.
Relevant: https://stackoverflow.com/a/45020525